sound/soc/sof/intel/pm_d0ix.c

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: (GPL-2.0 OR BSD-3-Clause)
+//
+// This file is provided under a dual BSD/GPLv2 license.  When using or
+// redistributing this file, you may do so under either license.
+//
+// Copyright(c) 2019 Intel Corporation. All rights reserved.
+//
+// Author: Keyon Jie <yang.jie@linux.intel.com>
+//
+
+#include <linux/device.h>
+#include <linux/module.h>
+#include <linux/platform_device.h>
+#include <linux/pm_runtime.h>
+#include <sound/sof.h>
+#include "../ops.h"
+
+/* time in ms for D0I3 runtime suspend delay */
+#define SND_SOF_D0I3_DELAY_MS	5000
+
+#ifdef CONFIG_PM
+static int sof_d0ix_suspend(struct device *dev)
+{
+	struct snd_sof_dev *sdev = dev_get_platdata(dev);
+
+	dev_dbg(dev, "Suspending to D0i3...\n");
+
+	return snd_sof_set_dsp_power_state(sdev, SOF_DSP_D0I3);
+}
+
+static int sof_d0ix_resume(struct device *dev)
+{
+	struct snd_sof_dev *sdev = dev_get_platdata(dev);
+
+	dev_dbg(dev, "Resuming from D0i3...\n");
+
+	return snd_sof_set_dsp_power_state(sdev, SOF_DSP_D0I0);
+}
+#else
+#define sof_d0ix_suspend NULL
+#define sof_d0ix_resume NULL
+#endif
+
+static int pm_d0ix_probe(struct platform_device *pdev)
+{
+	/* allow runtime_pm */
+	pm_runtime_set_autosuspend_delay(&pdev->dev, SND_SOF_D0I3_DELAY_MS);
+	pm_runtime_use_autosuspend(&pdev->dev);
+
+	pm_runtime_allow(&pdev->dev);
```

Comment:
@keyonjie this is not needed for platform devices only for PCI devices. pm_runtime_enable() will do the job